File: android/ccda-engine/src/main/kotlin/com/ccdakit/engine/Models/Common/CCDAIdentifier.kt
package com.ccdakit.engine

/**
 * CDA identifier with optional root and extension components.
 *
 * @property root Identifier root, usually an OID.
 * @property extensionValue Identifier extension value.
 */
data class CCDAIdentifier(
    val root: String? = null,
    val extensionValue: String? = null,
) {
    /** Root and extension combined for lightweight display and stable comparisons. */
    val stringValue: String
        get() = listOfNotNull(root, extensionValue).joinToString(" / ")
}

File: apple/Sources/CCDAEngine/Models/Common/CCDAIdentifier.swift
import Foundation

/// C-CDA identifier represented by root and optional extension.
public struct CCDAIdentifier: Hashable, Sendable {
    /// Identifier root, often an OID.
    public let root: String?
    /// Identifier extension value.
    public let extensionValue: String?

    /// Creates an identifier value.
    public init(root: String? = nil, extensionValue: String? = nil) {
        self.root = root
        self.extensionValue = extensionValue
    }

    /// Identifier text combining root and extension.
    public var stringValue: String {
        [root, extensionValue].compactMap { $0 }.joined(separator: " / ")
    }
}
